modules/serving-transport-encryption-overview.adoc

```diff
@@ -0,0 +1,56 @@
+
+. *Cluster-local encryption*: Transport encryption on the ingress layer internal to the cluster, for example cluster-local domains such as `myapp.<namespace>.svc.cluster.local`.
+
+. *System-internal encryption*: Transport encryption between Knative internal components: `ingress-gateway`, `activator`, `queue-proxy`.
+
+[IMPORTANT]
+====
+All control-plane traffic, including Kubernetes PreStopHooks, metadata, and metrics, is not yet encrypted. As such traffic contains no user data, its encryption is considered to be implemented in a future release.
+====
+
+The different parts are independent of each other and can be enabled and disabled individually. Also, they can use the same or different Certificate Authorities (CAs) to sign the necessary certificates.
+
+[id="serving-transport-encryption-external-domain_{context}"]
+== External domain encryption
+
+The transport encryption for external domains is handled by the ingress layer of the cluster, which is either the OpenShift ingress or OpenShift Service Mesh.
+
+[id="serving-transport-encryption-cluster-local-encryption_{context}"]
+== Cluster-local encryption
+
+Cluster-local encryption enables transport encryption for cluster-local domains. It has the following properties:
+
+* Certificate Common Name (CN) or Subject Alternative Name (SAN) contains the cluster-local domains of a Knative Service, for example `myapp.namespace.svc.cluster.local`, `myapp.namespace.svc`, `myapp.namespace`.
+
+* The certificates are selected using SNI by the cluster-local endpoint of `ingress-controller`.
+
+* The caller has to trust the CA that signed the cluster-local certificates. This is out of the scope of OpenShift Serverless.
+
+* To create the certificates, Knative relies on `cert-manager`, which needs to be installed and configured for the feature to work.
+
+[id="serving-transport-encryption-system-internal-encryption_{context}"]
+== System-internal encryption
+
+System-internal encryption enables transport encryption for system internal components: Ingress-Gateway, Activator, and Queue-Proxy. These components are hosting TLS endpoints when this configuration is used. Note the following:
+
+* To get the certificates, {ServerlessProductName} relies on `cert-manager`, which needs to be installed and configured for the feature to work.
+
+* Specific SANs are used to verify each connection. Each component needs to trust the CA that signed the certificates. For this, {ServerlessProductName} system components will consume and trust a bundle of CAs. The CA bundle needs to be provided by the cluster administrator.
+
+[id="additional-resources_serving-transport-encryption-overview"]
+[role="_additional-resources"]
+== Additional resources
+* link:https://openshift-knative.github.io/docs/docs/latest/serverless/serving/serving-transport-encryption.html[OpenShift Serverless Serving Transport Encryption]
```

Comment:
Do we want to include this link? It points to Tech Preview docs. I believe we're now converting the TP docs into GA docs in this PR. 
The TP docs still include some information that is not included in the current PR, for example the pictures/graphs of communication. Is it intentional that we don't include them in the current PR?

Reply:
@mgencur In the docs, all graphics need to be done by a dedicated person -- to confirm to a unified style. The current queue for the graphics is very long, so if we waited for them, these docs wouldn't be published in 2024.

For this reason, we decided to link to these TP docs with a mention that the accompanying diagrams are there.

Reply:
Including links to Github repos from our product docs requires an exception from PX. See:
- https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#links-to-external-websites
- https://docs.google.com/document/d/1WN7k72PKxcPA__erp5TK-CefmS_sVwE3r40PvpmNk3E/edit#bookmark=id.woxoarp6tluj

If there's no specific PX person for serverless, it probably needs to be OK'd by @sferich888 

Max is correct about the process for requesting diagrams - information is available [here](https://source.redhat.com/groups/public/diagrams). However, rather than linking to the diagrams, I would put in a ticket for the RH branded ones and open a SRVLS docs Jira to follow up and add these properly once they're ready.

Reply:
There was a firm request from Naina to have at least a link to these diagrams. The theoretical part of the docs indeed loses a lot without the diagrams. The ticket for RH branded ones has been created a while ago, and will take the diagram team a much longer while to complete.